### PR TITLE
Login

### PR DESCRIPTION
**Pull request by jbcaud**
Pull request to solve #5. Created login/logout functionality, as well as basic navigation for login and registration. Also added error handling for login using a snackbar. Attached is a video demo of these updates.

https://user-images.githubusercontent.com/54962983/155049533-c98d701d-84cf-4bb1-a5f8-febf3eed0f34.mov

### CODE REVIEW

**Comment by usjpin** [comment]
PR closes #5

File: frontend/constants.js [hunk 1]
```diff
@@ -20,16 +21,22 @@ export const UI = {
     },
 };
 
-const baseURL = "http://localhost:8000/v1/";
+const baseURL = "http://100.64.1.122:8000/v1/";
```
Comment (usjpin) [inline frontend/constants.js:24]: you might wanna remove this

File: frontend/services/UserService.js [hunk 2]
```diff
@@ -18,6 +18,25 @@ const UserService = {
...
+            }),
+        });
+        return res;
+    },
+    logout: async function () {
+        const res = await fetch(Network.Requests.URLS.logout, {
+            method: "POST",
+            headers: Network.Requests.BasicHeaders
+        });
+
```
Comment (usjpin) [inline frontend/services/UserService.js:37]: inconsistent new line to login (nitpick)

File: frontend/components/Login.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+import Spacer from "./reusables/Spacer";
+import TextField from "./reusables/TextField";
+import CustomSnackBar from "./reusables/CustomSnackBar";
+
+const Login = ({
+    navigation,
+    route: {
+    },
+}) => {
+    const [visible, setVis] = useState(false);
```
Comment (usjpin) [inline frontend/components/Login.js:20]: please say "setVisible" instead to "setVis"

**Review by usjpin — CHANGES_REQUESTED** [review]
some nitpicks

File: frontend/.gitignore [hunk 4]
```diff
@@ -2,5 +2,8 @@
...
 
 # Deployment files
-.expo
-.expo-shared
\ No newline at end of file
+*.expo
+*.expo-shared
+
+# Error logs
+*.log
```
Comment (usjpin) [inline frontend/.gitignore:9]: from jacob to jacob: remove the log file

File: frontend/components/Login.js [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+                </Card>
+                <Spacer />
+                <Spacer />
+                <PrimaryButton onPress={login}>
+                    <ButtonText text="Login" />
+                </PrimaryButton>
+                <Spacer />
+                <Text style = {{ color: 'yellow', fontSize: 24 }} onPress = {() => navigation.navigate(Navigation.Landing)}>
+                    Don't have an account?
+                </Text>
```
Comment (drkuster) [inline frontend/components/Login.js:92]: 1.) I think this should navigate to registration 2.) Not rlly sure about the yellow color. But if we're going to use yellow consistently throughout the app then it should be added to the constants.

File: frontend/components/Registration.js [hunk 6]
```diff
@@ -30,6 +30,8 @@ const Registration = ({
             newUser.password
         ).then((response) => {
             console.log(response);
+        }).catch((e) => {
+            console.log("Error: ", e);
```
Comment (drkuster) [inline frontend/components/Registration.js:34]: Thxx for this

Comment (drkuster) [reply frontend/components/Registration.js:34]: If you end up snackbarring the login error, could you snackbar this as well as a big favor

**Review by drkuster — CHANGES_REQUESTED** [review]

File: frontend/components/Login.js (hunk 5 above)
Comment (drkuster) [reply frontend/components/Login.js:92]: Ignore this for now until / if the landing page ends up getting reformatted

File: frontend/services/UserService.js (hunk 2 above)
Comment (jbcaud) [reply frontend/services/UserService.js:37]: fixed in new commit

File: frontend/components/Login.js (hunk 3 above)
Comment (jbcaud) [reply frontend/components/Login.js:20]: fixed in newest commit

File: frontend/.gitignore (hunk 4 above)
Comment (jbcaud) [reply frontend/.gitignore:9]: fixed in newest commit. thank you jacob!

File: frontend/components/Landing.js [hunk 7]
```diff
@@ -6,17 +6,25 @@ import Heading from "./reusables/Heading";
...
     const register = (userType) => {
         navigation.navigate(Navigation.Registration, {
             userType: userType,
         });
     };
+    const containerStyle = StyleSheet.create({
+        container: {
+            top: -35
+        }
+    });
```
[the remark this answers is not in the document]
Comment (jbcaud) [reply frontend/components/Landing.js:22]: fixed in newest commit

File: frontend/components/Login.js [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+    navigation,
+    route: {
+    },
+}) => {
+    const [visible, setVis] = useState(false);
+    const [message, setMessage] = useState("");
+    const [currUser, setState] = useState({
+        email: "",
+        password: "",
+    });
```
[the remark this answers is not in the document]
Comment (usjpin) [reply frontend/components/Login.js:25]: From a React perspective, it’s more efficient to keep them separate as React would only render the components that are dependent on those hooks. Also you might need individual dependencies for useEffect() n stuff so I think this approach is fine.

**Review by usjpin — APPROVED** [review]

**Review by drkuster — APPROVED** [review]

**Pull request merged**